src/tasks/dump.ts

```diff
@@ -0,0 +1,600 @@
+ * - handling the case where the receiver is not the signer address
+ * - handling ETH buy addresses (especially with a custom receiver)
+ * - rejecting dump requests for tokens for which the fee to pay is larger than
+ *   a given threshold
+ */
+async function getDumpInstructions({
+  dumpedTokens: inputDumpedTokens,
+  toTokenAddress,
+  user,
+  allowanceManager,
+  maxFeePercent,
+  hasCustomReceiver,
+  hre,
+  network,
+  api,
+}: GetDumpInstructionInput): Promise<DumpInstructions> {
+  // todo: support dumping ETH by wrapping them
+  if (inputDumpedTokens.includes(BUY_ETH_ADDRESS)) {
+    throw new Error(
+      `Dumping the native token is not supported. Remove the ETH flag address ${BUY_ETH_ADDRESS} from the list of tokens to dump.`,
+    );
+  }
+
+  let toToken: Erc20Token | NativeToken;
+  if (toTokenAddress === undefined || toTokenAddress === BUY_ETH_ADDRESS) {
+    toToken = nativeToken(hre);
+  } else {
+    const erc20 = await erc20Token(toTokenAddress, hre);
+    if (erc20 === null) {
+      throw new Error(
+        `Input toToken at address ${toTokenAddress} is not a valid Erc20 token.`,
+      );
+    }
+    toToken = erc20;
+  }
+
+  let transferToReceiver: TransferToReceiver | undefined = undefined;
+  const dumpedTokens = Array.from(new Set(inputDumpedTokens)).filter(
+    (token) => token !== (toTokenAddress ?? BUY_ETH_ADDRESS),
+  );
+  if (
+    hasCustomReceiver &&
+    inputDumpedTokens.includes(toTokenAddress ?? BUY_ETH_ADDRESS)
+  ) {
+    transferToReceiver = {
+      token: toToken,
+      amount: await balanceOf(toToken, user),
+    };
+  }
+
+  const computedInstructions: (DumpInstruction | null)[] = (
+    await promiseAllWithRateLimit(
+      dumpedTokens.map((tokenAddress) => async ({ consoleWarn }) => {
+        const token = await erc20Token(tokenAddress, hre);
+        if (token === null) {
+          consoleWarn(
+            `Dump request skipped for invalid ERC-20 token at address ${tokenAddress}.`,
+          );
+          return null;
+        }
+        const [balance, approvedAmount]: [
+          BigNumber,
+          BigNumber,
+        ] = await Promise.all([
+          token.contract.balanceOf(user).then(BigNumber.from),
+          token.contract.allowance(user, allowanceManager).then(BigNumber.from),
+        ]);
+        const needsAllowance = approvedAmount.lt(balance);
+        if (balance.isZero()) {
+          consoleWarn(
+            `Dump request skipped for token ${displayName(
+              token,
+            )}. No balance for that token is available.`,
+          );
+          return null;
+        }
+        const sellToken = token.address;
+        const buyToken = isNativeToken(toToken)
+          ? WRAPPED_NATIVE_TOKEN_ADDRESS[network] // todo: replace WETH address with BUY_ETH_ADDRESS when services support ETH estimates
+          : toToken.address;
+        const kind = OrderKind.SELL;
+        const fee = await api.getFee({
```

Comment:
Maybe we should already use the new "combined" endpoint here: https://protocol-mainnet.gnosis.io/api/#/default/get_api_v1_feeAndQuote_sell